### PR TITLE
feat: Nav-Bar Added(#113 #143)

### LINKED ISSUE #113: feat: Home Page for LinkFree Project

### PR DESCRIPTION
**Pull request by Soham041201**
Fixes #113 #143 

After a long discussion and rebase and commits. Creating a fresh PR containing a simple nav bar added a bit of margin to make the links look a bit better

## Screenshot
Screenshot from 2021-09-19 21-07-37

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/249"></a>

### CODE REVIEW

**Review by CodingSpecies — APPROVED** [review]
LGTM

**Review by krishdevdb — APPROVED** [review]
Seems good to me 👍🏻

**Review by AbhijithGanesh — CHANGES_REQUESTED** [review]
Line 41, in Home.js , Elements are not terminated subsequently failing checks, Please fix them

**Review by AbhijithGanesh — CHANGES_REQUESTED** [review]
Remove 
```js
import { useHistory } from 'react-router-dom'
```
since it isn't being used.

File: src/index.css [hunk 1]
```diff
@@ -31,3 +31,15 @@ code {
...
   text-align: center;
 }
+
+.navbar a{
+  text-decoration: none;
+  color: skyblue;
+  font-size: 18px;;
+  padding-right: 20px;
+}
+.navbar{
```
Comment (Panquesito7) [inline src/index.css:41]:
```suggestion

.navbar {
```

File: src/index.css [hunk 2]
```diff
@@ -31,3 +31,15 @@ code {
...
+margin-bottom: 100px;
+background: white;
+
```
Comment (Panquesito7) [inline src/index.css:44]:
```suggestion
```

File: src/index.css [hunk 3]
```diff
@@ -31,3 +31,15 @@ code {
...
+.navbar a{
+  text-decoration: none;
+  color: skyblue;
+  font-size: 18px;;
+  padding-right: 20px;
+}
+.navbar{
+margin-bottom: 100px;
+background: white;
```
Comment (Panquesito7) [inline src/index.css:43]:
```suggestion
  margin-bottom: 100px;
  background: white;
```

File: src/index.css [hunk 4]
```diff
@@ -31,3 +31,15 @@ code {
 .align-center {
   text-align: center;
 }
+
+.navbar a{
```
Comment (Panquesito7) [inline src/index.css:35]:
```suggestion
.navbar a {
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: src/index.css (hunk 1 above)
Comment (Soham041201) [reply src/index.css:41]: What is the issue, if we do not add a space?

**Review by Panquesito7 — COMMENTED** [review]
The navbar looks good, but I recommend we should improve it and make it look better. Just an idea, it doesn't have to be implemented. 🙂
Let us see what other maintainers think about this.

**Review by CodingSpecies — CHANGES_REQUESTED** [review]
@Soham041201 please can you resolve the conflicts? Thank you.

**Comment by Soham041201** [comment]
> @Soham041201 please can you resolve the conflicts? Thank you.

Removed :)

**Review by Soham041201 — COMMENTED** [review]
Error solved

File: src/index.css [hunk 5]
```diff
@@ -11,3 +11,43 @@ code {
...
+}
+
+.align-center {
+  text-align: center;
+}
+
+.navbar a {
+  text-decoration: none;
+  color: skyblue;
+  font-size: 18px;;
```
Comment (eddiejaoude) [inline src/index.css:39]: double `;`

File: src/index.css [hunk 6]
```diff
@@ -11,3 +11,43 @@ code {
...
+  transition: transform 0.33s !important;
+}
+
+.p-button:hover {
+  transform: scale(1.02);
+}
+
+.p-avatar {
+  transition: transform 0.33s !important;
+}
```
Comment (eddiejaoude) [inline src/index.css:26]: These I think have now be moved to their relevant components

File: src/Components/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Menubar } from 'primereact/menubar'
+export default function Navbar() {
+  return (
+    <div className="navbar">
+      <Menubar
+        start={<h2>LinkFree</h2>}
+        end={<a href="#">Home</a>}
```
Comment (eddiejaoude) [inline src/Components/Navbar.js:8]: I think this will need a path? `/`?

File: src/index.css [hunk 8]
```diff
@@ -11,3 +11,43 @@ code {
...
+
+.p-avatar:hover {
+  transform: scale(1.2);
+}
+
+.align-center {
+  text-align: center;
+}
```
Comment (eddiejaoude) [inline src/index.css:34]: PrimeReact has a class already for this I believe

File: src/index.css [hunk 9]
```diff
@@ -11,3 +11,43 @@ code {
...
+  color: skyblue;
+  font-size: 18px;;
+  padding-right: 20px;
+}
+
+.navbar {
+  margin-bottom: 100px;
+  background: white;
+}
+.w-50 {
```
Comment (eddiejaoude) [inline src/index.css:47]: Inconsistent spacing between classes

File: src/index.css [hunk 10]
```diff
@@ -11,3 +11,43 @@ code {
...
+  max-width: 50vw;
+  margin: 10px auto;
+  text-align: center;
+  line-height: 30px;
```
Comment (eddiejaoude) [inline src/index.css:51]: Can these be achieve using PrimeReact pre-made classes

**Comment by eddiejaoude** [comment]
Great collaboration everyone, I left a few comments too

File: src/index.css (hunk 10 above)
Comment (Soham041201) [reply src/index.css:34]: I did try with it wasnt able to perform. I thoght too push a basic skeleton and then get into css properties

**Comment by Soham041201** [comment]
> Great collaboration everyone, I left a few comments too

Made the changes. Please review it once :)

**Review by qwe123coder — APPROVED** [review]
LGTM :rocket:

File: src/index.css (hunk 10 above)
Comment (eddiejaoude) [reply src/index.css:34]: Did you look at the NavBar in PrimeReact? Looks like it has a component with this all done and then we can focus on adding features to it https://www.primefaces.org/primereact/showcase/#/menubar

Screenshot 2021-09-29 at 11 07 01

Comment (Soham041201) [reply src/index.css:34]: Okay..I will simply add this one ...then we can think which parts should be removed and which should be kept

**Comment by Soham041201** [comment]
Made the changes as requested by @eddiejaoude . Added a model Menu Bar. Up for modification and development

**Comment by Soham041201** [comment]
image

**Comment by eddiejaoude** [comment]
This PR has so many changes it is unclear. @Soham041201 I think it is best to create new PRs with only specific changes in each PR

**Comment by Soham041201** [comment]
> This PR has so many changes it is unclear. @Soham041201 I think it is best to create new PRs with only specific changes in each PR

Yes!! Was going to do this yesterday itself. Also this time it will count as a PR for hactober fest :)

**Comment by eddiejaoude** [comment]
did a new PR get created?

**Comment by Soham041201** [comment]
> did a new PR get created?

Yess Please checkout PR: #425

**Pull request closed without merging**